R/rowClust.R

```diff
@@ -0,0 +1,83 @@
+#' Clusters rowData using \code{fastcluster::hclust}
+#'
+#' \code{rowClust} Performs hierarchical clustering on rowData of the 
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}, 
+#' by using the dissimilarity function provided at \code{dissimilarity_FUN}.
+#'
+#' @param x a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param abund_values a \code{character} vector, name of the assay containing the data.
+#' 
+#' @param name a \code{character} vector, name of the new column at \code{rowData} where to place cluster ids.
+#' 
+#' @param clustering_method a \code{character} vector representing the clustering 
+#' method to be used, same as methods available at \code{\link[hclust]{hclust}}.
+#' 
+#' @param transposed \code{logical} whether the data should be transposed.
+#' 
+#' @param dissimilarity_FUN \code{stats::cor} function to compute the dissimilarity matrix
+#' 
+#' @param ... other arguments passed onto \code{\link[cutree]{cutree}}
+#'
+#' @return \code{x} with additional \code{\link{rowData}} named \code{*name*}
+#'
+#' @export
+#'
+#' @examples
+#' # Load example data
+#' library(mia)
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
+#' # Agglomerate to Class rank; for speeding up the example.
+#' tse <- agglomerateByRank(tse, rank="Class")
+#' # Cluster using Spearman correlation as a dissimilarity and ward.D2 
+#' # for clustering; with results having 6 clusters
+#' tse <- rowClust(tse, clustering_method = "ward.D2",
+#'                 dissimilarity_FUN = function(x) stats::cor(x, method="spearman"),
```

Comment:
Important: `stats::cor` is similarity method (increases with similarity) and not dissimilarity method (decreases with similarity). The common metrics, like Euclidean distance, are dissimilarity methods. The treatment should be uniform, and here the corresponding dissimilarity function should be `1 - stats::cor`.  Fix.